### PR TITLE
Add llama.cpp docs

### PR DESCRIPTION
**Pull request by ngxson**

### CODE REVIEW

**Comment by ngxson** [comment]
## TODO: revert commit fb60e6e before merging

**Review by Michellehbn — APPROVED** [review]

File: docs/source/others/llamacpp_container.mdx [hunk 1]
```diff
@@ -0,0 +1,87 @@
+# Deploying a llama.cpp Container
+
+When creating an endpoint with a [GGUF](https://huggingface.co/docs/hub/en/gguf) model, a [llama.cpp](https://github.com/ggerganov/llama.cpp) container is automatically selected, utilizing the latest image built from the `master` branch of the llama.cpp repository. Upon deployment, a server with an OpenAI-compatible endpoint becomes available.
```
Comment (osanseviero) [inline docs/source/others/llamacpp_container.mdx:3]:
```suggestion
When creating an endpoint with a [GGUF](https://huggingface.co/docs/hub/en/gguf) model, a [llama.cpp](https://github.com/ggerganov/llama.cpp) container is automatically selected using the latest image built from the `master` branch of the llama.cpp repository. Upon deployment, a server with an OpenAI-compatible endpoint becomes available.
```

File: docs/source/others/llamacpp_container.mdx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+Alternatively, you can follow the video tutorial below for a step-by-step guide on deploying an endpoint with a llama.cpp container:
+
+<video width="1280" height="720" controls>
+  <source src="https://raw.githubusercontent.com/huggingface/hf-endpoints-documentation/refs/heads/xsn/llamacpp/assets/llamacpp_guide_video.mp4" type="video/mp4">
+</video>
+
+## Configurations
+
+The llama.cpp container offers several configuration options that can be adjusted. After deployed, you can modify these settings by accessing the **Settings** tab in the endpoint details page.
```
Comment (osanseviero) [inline docs/source/others/llamacpp_container.mdx:35]:
```suggestion
The llama.cpp container offers several configuration options that can be adjusted. After deployment, you can modify these settings by accessing the **Settings** tab on the endpoint details page.
```

File: docs/source/others/llamacpp_container.mdx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+- `LLAMA_ARG_NO_MMAP`
+- `LLAMA_ARG_CTX_SIZE`
+- `LLAMA_ARG_N_PARALLEL`
+- `LLAMA_ARG_ENDPOINT_METRICS`
+
+## Troubleshooting
+
+In case the deployment fails, please watch the log output for any error messages.
+
+You can access the logs by clicking on the **Logs** tab in the endpoint details page. To learn more, refer to the [Logs](../guides/logs) documentation.
```
Comment (osanseviero) [inline docs/source/others/llamacpp_container.mdx:63]:
```suggestion
You can access the logs by clicking on the **Logs** tab on the endpoint details page. To learn more, refer to the [Logs](../guides/logs) documentation.
```

File: docs/source/others/llamacpp_container.mdx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+```
+ggml_backend_cuda_buffer_type_alloc_buffer: allocating 67200.00 MiB on device 0: cuda
+Malloc failed: out of memory
+llama_kv_cache_init: failed to allocate buffer for kv cache
+llama_new_context_with_model: llama_kv_cache_init() failed for self-attention cache
+...
+```
+
+That means the selected hardware configuration does not have enough memory to accommodate the selected GGUF model. You can try:
```
Comment (osanseviero) [inline docs/source/others/llamacpp_container.mdx:77]:
```suggestion
That means the selected hardware configuration does not have enough memory to accommodate the selected GGUF model. You can try to:
```

**Review by osanseviero — APPROVED** [review]
Nice 🔥

File: docs/source/others/llamacpp_container.mdx [hunk 5]
```diff
@@ -0,0 +1,87 @@
+# Deploying a llama.cpp Container
+
+When creating an endpoint with a [GGUF](https://huggingface.co/docs/hub/en/gguf) model, a [llama.cpp](https://github.com/ggerganov/llama.cpp) container is automatically selected using the latest image built from the `master` branch of the llama.cpp repository. Upon deployment, a server with an OpenAI-compatible endpoint becomes available.
```
Comment (Vaibhavs10) [inline docs/source/others/llamacpp_container.mdx:3]:
```suggestion
You can deploy any llama.cpp compatible GGUF on the Hugging Face Endpoints. When you create an endpoint with a [GGUF](https://huggingface.co/docs/hub/en/gguf) model, a [llama.cpp](https://github.com/ggerganov/llama.cpp) container is automatically selected using the latest image built from the `master` branch of the llama.cpp repository. Upon successful deployment, a server with an OpenAI-compatible endpoint becomes available.
```

File: docs/source/others/llamacpp_container.mdx [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+For a comprehensive list of available endpoints, please refer to the [API documentation](https://github.com/ggerganov/llama.cpp/blob/master/examples/server/README.md#api-endpoints).
```
Comment (Vaibhavs10) [inline docs/source/others/llamacpp_container.mdx:5]:
```suggestion
Llama.cpp supports multiple endpoints like `/tokenize`, `/health`, `/embedding` and many more. For a comprehensive list of available endpoints, please refer to the [API documentation](https://github.com/ggerganov/llama.cpp/blob/master/examples/server/README.md#api-endpoints).
```

File: docs/source/others/llamacpp_container.mdx [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+
+2. Choose the desired GGUF file, noting that memory requirements will vary depending on the selected file. For example, an F16 model requires more memory than a Q4_K_M model.
+
+<img src="https://raw.githubusercontent.com/huggingface/hf-endpoints-documentation/refs/heads/xsn/llamacpp/assets/llamacpp_2.png" alt="Select GGUF file" />
+
+3. Select your desired hardware configuration.
+
+<img src="https://raw.githubusercontent.com/huggingface/hf-endpoints-documentation/refs/heads/xsn/llamacpp/assets/llamacpp_3.png" alt="Select hardware" />
+
+4. Optionally, you can customize the container's configuration settings. For more information, please refer to the **Configurations** section below.
```
Comment (Vaibhavs10) [inline docs/source/others/llamacpp_container.mdx:23]:
```suggestion
4. Optionally, you can customize the container's configuration settings like `Max Tokens`, `Number of Concurrent Requests`. For more information on those, please refer to the **Configurations** section below.
```

File: docs/source/others/llamacpp_container.mdx [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+
+5. Click the **Create Endpoint** button to complete the deployment.
+
+Alternatively, you can follow the video tutorial below for a step-by-step guide on deploying an endpoint with a llama.cpp container:
```
Comment (Vaibhavs10) [inline docs/source/others/llamacpp_container.mdx:27]: Nice! we can switch this to a public HF dataset possibly

File: docs/source/others/llamacpp_container.mdx [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+</video>
+
+## Configurations
+
+The llama.cpp container offers several configuration options that can be adjusted. After deployment, you can modify these settings by accessing the **Settings** tab on the endpoint details page.
+
+### Basic Configurations
+
+- **Max Tokens (per Request)**: The maximum number of tokens that can be sent in a single request.
+- **Max Concurrent Requests**: The maximum number of concurrent requests allowed for this deployment. Increasing this limit requires additional memory allocation. For instance, setting this value to 4 requests with 1024 tokens maximum per request requires memory capacity for 4096 tokens in total.
```
Comment (Vaibhavs10) [inline docs/source/others/llamacpp_container.mdx:40]:
```suggestion
- **Max Concurrent Requests**: The maximum number of concurrent requests allowed for this deployment. Increasing this limit requires additional memory allocation. 
For instance, setting this value to 4 requests with 1024 tokens maximum per request requires memory capacity for 4096 tokens in total.
```

Adding a line break here would make it more readable.

File: docs/source/others/llamacpp_container.mdx [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+- `LLAMA_ARG_N_GPU_LAYERS`
+- `LLAMA_ARG_EMBEDDINGS`
+- `LLAMA_ARG_HOST`
+- `LLAMA_ARG_PORT`
+- `LLAMA_ARG_NO_MMAP`
+- `LLAMA_ARG_CTX_SIZE`
+- `LLAMA_ARG_N_PARALLEL`
+- `LLAMA_ARG_ENDPOINT_METRICS`
+
+## Troubleshooting
```
Comment (Vaibhavs10) [inline docs/source/others/llamacpp_container.mdx:59]: Can we convert this into a bullet point list?

**Review by Vaibhavs10 — COMMENTED** [review]
Nice and thanks for picking this up, left some wording suggestions.

File: docs/source/others/llamacpp_container.mdx (hunk 10 above)
Comment (ngxson) [reply docs/source/others/llamacpp_container.mdx:60]: Done:
image

**Pull request merged**